pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -255,6 +265,42 @@ type TriggerParameterSource struct {
 	Value *string `json:"value,omitempty" protobuf:"bytes,3,opt,name=value"`
 }
 
+// TriggerPolicy dictates the policy for the trigger retries
+type TriggerPolicy struct {
+	// Backoff before checking resource state
+	Backoff Backoff `json:"backoff" protobuf:"bytes,1,opt,name=backoff"`
+
+	// Criteria to check resource state
+	Criteria *TriggerPolicyCriteria `json:"criteria" protobuf:"bytes,2,opt,name=criteria"`
+
+	// ErrorOnBackoffTimeout determines whether sensor should transition to error state if the backoff times out and yet the resource neither transitioned into success or failure.
+	ErrorOnBackoffTimeout bool `json:"errorOnBackoffTimeout" protobuf:"bytes,3,opt,name=errorOnBackoffTimeout"`
```

Comment:
Is there any case that NO error on timeout is desired?

Reply:
I guess it is useful when a failed trigger doesn't necessarily mean that next one can't proceed. e.g. If you have a trigger that basically writes to a S3 bucket and the second trigger expects the data to be available in the bucket but also has a fallback mechanism. In that case, you would want to set back off on the first trigger to wait till it goes in success state. If after backoff, it is not possible to figure out the state of the triggered resource or it failed, you still want second trigger object to get created